Error when building portainer locally with Yarn

I'm seeing the following error when I run `yarn start` on an Ubuntu 18.04 x64 machine:

_github.com/portainer/libcompose (download)
# github.com/portainer/portainer/api/cron
../../cron/scheduler.go:22:2: too many arguments to return
	have (cron.EntryID, error)
	want (error)
../../cron/scheduler.go:38:8: assignment mismatch: 1 variable but newCron.AddJob returns 2 values
../../cron/scheduler.go:72:8: assignment mismatch: 1 variable but newCron.AddJob returns 2 values
mv: cannot stat 'portainer': No such file or directory
 Use --force to continue.

Aborted due to warnings.
error Command failed with exit code 6._

I have verified that Docker, Node.js (8.10.0), yarn and Golang (1.12.7) are installed on my system. 

Steps to reproduce:

1. `git clone https://github.com/portainer/portainer.git`
2. `mkdir -p ${GOPATH}/src/github.com/portainer`
3. `ln -s ${PWD}/portainer ${GOPATH}/src/github.com/portainer/portainer`
4. `cd portainer`
5. `yarn`
6. `yarn start`

I tried to run the following and got the same error:
`cd api/cmd/portainer`
`CGO_ENABLED=0 go build -a --installsuffix cgo --ldflags '-s' && mv portainer ../../../dist`

Output:

`# github.com/portainer/portainer/api/cron
../../cron/scheduler.go:22:2: too many arguments to return
	have (cron.EntryID, error)
	want (error)
../../cron/scheduler.go:38:8: assignment mismatch: 1 variable but newCron.AddJob returns 2 values
../../cron/scheduler.go:72:8: assignment mismatch: 1 variable but newCron.AddJob returns 2 values`

Based on [Issue 2727](https://github.com/portainer/portainer/issues/2727), I triple checked my `~/.profile` to make sure `GOPATH` was set correctly. This is what it shows:

`# ~/.profile: executed by Bourne-compatible login shells.

if [ "$BASH" ]; then
  if [ -f ~/.bashrc ]; then
    . ~/.bashrc
  fi
fi

mesg n || true

export GOROOT=$HOME/go
export GOPATH=$HOME/work
export PATH=$PATH:$GOROOT/bin:$GOPATH/bin`

I have previously built portainer.io locally with yarn before. I'm not sure what I'm doing differently, or incorrectly.

I was going to ask this on Slack but I couldn't find a way to request for an invitation.

--- a/api/cron/scheduler.go
+++ b/api/cron/scheduler.go
@@ -19,7 +19,8 @@ func NewJobScheduler() *JobScheduler {
 
 // ScheduleJob schedules the execution of a job via a runner
 func (scheduler *JobScheduler) ScheduleJob(runner portainer.JobRunner) error {
-	return scheduler.cron.AddJob(runner.GetSchedule().CronExpression, runner)
+	_, err := scheduler.cron.AddJob(runner.GetSchedule().CronExpression, runner)
+	return err
 }
 
 // UpdateSystemJobSchedule updates the first occurence of the specified
@@ -35,7 +36,7 @@ func (scheduler *JobScheduler) UpdateSystemJobSchedule(jobType portainer.JobType
 
 	for _, entry := range cronEntries {
 		if entry.Job.(portainer.JobRunner).GetSchedule().JobType == jobType {
-			err := newCron.AddJob(newCronExpression, entry.Job)
+			_, err := newCron.AddJob(newCronExpression, entry.Job)
 			if err != nil {
 				return err
 			}
@@ -69,7 +70,7 @@ func (scheduler *JobScheduler) UpdateJobSchedule(runner portainer.JobRunner) err
 				jobRunner = entry.Job
 			}
 
-			err := newCron.AddJob(runner.GetSchedule().CronExpression, jobRunner)
+			_, err := newCron.AddJob(runner.GetSchedule().CronExpression, jobRunner)
 			if err != nil {
 				return err
 			}

